src/cosim/slave_simulator.cpp

```diff
@@ -223,18 +225,17 @@ class set_variable_cache
                 const auto ref = entry.first;
                 if (exposedVariables_.at(ref).arrayIndex < 0) {
                     exposedVariables_.at(ref).arrayIndex = references_.size();
+                    back_refs_[ref] = references_.size();
                     assert(references_.size() == values_.size());
                     references_.emplace_back(ref);
                     values_.emplace_back(exposedVariables_.at(ref).lastValue);
                 }
-            }
-            assert(references_.size() == values_.size());
-            for (std::size_t i = 0; i < references_.size(); ++i) {
-                const auto iterator = modifiers_.find(references_[i]);
-                if (iterator != modifiers_.end()) {
-                    values_[i] = iterator->second(values_[i], deltaT);
+                const auto it = back_refs_.find(ref);
+                if (it != back_refs_.end()) {
+                    values_[it->second] = entry.second(values_[it->second], deltaT);
```

Comment:
I'm wondering about the need for `back_refs_`. Isn't the array index (`it->second` here) stored in `exposedVariables_.at(ref).arrayIndex`?

Reply:
I think you are right, adjusted accordingly.